### ISSUE TITLE
Cannot use require in script

### ISSUE THREAD

**Issue by greenstatic** [actions/github-script#6]
The way the function constructor is created, it is not possible to use `require` directly.
Calling `require` in your script will result in: `ReferenceError: require is not defined`. 

As a temporary hotfix, you can call the following before your `require` call: 
```js
const require = global.require || global.process.mainModule.constructor._load;
```
Solution was [adapted from here](https://stackoverflow.com/a/43492031).

It would be great if the function constructor was adapted to allow normal `require` calls.

**Comment by alexbde** [comment]
Another solution would be to define requirements as action parameter. For example:

    - uses: actions/github-script@master
      with:
        github-token: ${{ secrets.GITHUB_TOKEN }}
        require:
          - core: @actions/core
          - key: value
        script: |
          core.doSomething()
          key.doIt()

Defining for every key-value-pair something like following, so you can use `core` like `context`/`github`.

    const core = require('@actions/core')
    const key = require('value')

**Comment by alexbde** [comment]
Btw. @greenstatic do you have an idea, why your solution is not working in my case? See #10.

**Comment by greenstatic** [comment]
@alexbde I see you have already found #9 that fixes your issue. I suspect my workaround doesn't work in your case because `@actions/core` is an external dependency whereas in my case I used it to require `fs` which is bundled with Node.js.

I suspect the way the function for the script is run, it ignores the parent (`github-script`) `node_modules` directory which is where `@actions/core` is located. Consequently this means one can only require packages bundled with Node.js.

This however is just a theory which I have not verified.

**Comment by alexbde** [comment]
I see. Thank you for explanation :)

**Comment by jclem** [comment]
Hi! I missed a huge set of notifications being irresponsible with keyboard shortcuts 😆 I'm sorry! I'm looking into this and the open PR on this repo, as well.

**Comment by jclem** [comment]
I've just released version `0.3.0`, and `require` should be available! This does not solve the general dependency question @alexbde brings up, which I'll think about separately.

Thank you for opening this!
